core/adapters/eth_tx.go

```diff
@@ -41,23 +41,23 @@ func (e *EthTx) TaskType() models.TaskType {
 // Perform creates the run result for the transaction if the existing run result
 // is not currently pending. Then it confirms the transaction was confirmed on
 // the blockchain.
-func (etx *EthTx) Perform(input models.RunInput, store *strpkg.Store) models.RunOutput {
+func (e *EthTx) Perform(input models.RunInput, store *strpkg.Store) models.RunOutput {
 	if !store.TxManager.Connected() {
-		return pendingConfirmationsOrConnection(input)
+		return pendingOutgoingConfirmationsOrConnection(input)
 	}
 
-	if input.Status().PendingConfirmations() {
+	if input.Status().PendingOutgoingConfirmations() {
```

Comment:
Personally I think this is the wrong way to go about solving this problem, it'd be better to look up transactions corresponding to this Task, rather than relying on a special obscure status for this individual task. It's clearer, simpler and less error prone.

e.g:

```go
tx, err := FindTxBySurrogateID(null.StringFrom(input.JobRunID().String()))
if errors.Cause(err) == errors.NotFound {
  // logic for creating initialTX
} else if err != nil {
 // error job
}

// process existing attempts logic
```

Reply:
I can see your point, and it would make for significantly fewer changes to the code, but I'd offer a counterargument that Sam's approach has a few indirect benefits.  Curious what you think:

1. It decouples the system by ensuring that only one subsystem is responsible for determining which type of `PendingConfirmations` the run is in.  All other subsystems can just rely on the status handed to them.
    - If other subsystems end up needing to discern between the two, they don't need to reach out to the ORM for help.  Reaching out to the ORM involves I/O, and therefore more latency, which could (at least generally speaking) introduce races.
    - Separately, if a given subsystem only needs the ORM for this purpose and nothing else, we'll need to import the `orm` package.  I've seen some nasty import cycles pop up lately, and they can be pretty difficult to resolve cleanly, so I'm keen on avoiding them whenever possible.  Sure, you could make a one-method interface, but this is usually a sign of hastily-designed package architecture.

2. It might it easier to gather raw information from the DB about the state of a run for debugging if the status is explicitly recorded there.

Reply:
I'll elaborate more on what the benefits are of my approach, and we can weigh them up against eachother:

  1. Checking for an existing TxAttempt record is the safest way to ensure no double spends occur, although I don't believe that's possible in the code, thar be dragons and I can't be 100% certain. 
  2. The RunManager has to make special exceptions for this code that complicate it, every other adapter accepts the nominal "InProgress" state for task+job, I also sense there are ways this status can be overwritten away from `pending_confirmations` that this code doesn't handle and again ... lead to a double spend.
  3. It's true this will require more DB calls, but there is another ticket which should make this a moot point: https://www.pivotaltracker.com/story/show/169746497
  4. It won't require a migration, which could be risky, and cause unnecessary downtime.

In my opinion double spends are kind of a trump card, so I want to design away any chance for them to happen.

Reply:
Oh, and I forgot to mention: the suggestion to move most of this logic to the TxManager might solve some of the above ORM cycles/problems.

Reply:
I think we can go with a combination of both approaches.

Reply:
@j16r What about the `pending_connection` state. Do you think that ought to be eliminated as well?